File: webflux-dialogue/src/main/java/com/study/webflux/rag/domain/monitoring/port/MetricsRollupRepository.java
package com.study.webflux.rag.domain.monitoring.port;

import java.time.Instant;

import com.study.webflux.rag.domain.monitoring.model.MetricsGranularity;
import com.study.webflux.rag.domain.monitoring.model.MetricsRollup;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface MetricsRollupRepository {
	Mono<MetricsRollup> save(MetricsRollup rollup);

	Flux<MetricsRollup> findByGranularityAndBucketStartBetweenOrderByBucketStartAsc(
		MetricsGranularity granularity,
		Instant startTime,
		Instant endTime);
}

File: monitoring/src/main/kotlin/com/miyou/app/monitoring/port/MetricsRollupRepository.kt
package com.miyou.app.monitoring.port

import com.miyou.app.monitoring.model.MetricsGranularity
import com.miyou.app.monitoring.model.MetricsRollup
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono
import java.time.Instant

interface MetricsRollupRepository {
    fun save(rollup: MetricsRollup): Mono<MetricsRollup>

    fun findByGranularityAndBucketStartBetweenOrderByBucketStartAsc(
        granularity: MetricsGranularity,
        startTime: Instant,
        endTime: Instant,
    ): Flux<MetricsRollup>
}
